lib/matplotlib/axes/_base.py

```diff
@@ -2577,6 +2598,19 @@ def redraw_in_frame(self):
     def get_renderer_cache(self):
         return self._cachedRenderer
 
+    # layoutboxes: allow us to set these after instantiation.
+    def set_layoutbox(self, layoutbox):
+        """
+        Set the layoutbox for this axis...
+        """
+        self.layoutbox = layoutbox
+
+    def set_poslayoutbox(self, layoutbox):
+        """
+        Set the spinelayoutbox for this axis...
+        """
+        self.poslayoutbox = layoutbox
```

Comment:
Is there a reason for using setters instead of directly assigning to the attributes?  This is related to my question about whether the attributes are part of the API.